pkg/grafana/alertgroup-handler.go

```diff
@@ -0,0 +1,210 @@
+package grafana
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"encoding/json"
+	"errors"
+
+	"github.com/grafana/grizzly/pkg/grizzly"
+	"github.com/grafana/tanka/pkg/kubernetes/manifest"
+
+	"github.com/grafana/grafana-openapi-client-go/client/provisioning"
+	"github.com/grafana/grafana-openapi-client-go/models"
+)
+
+// AlertRuleGroupHandler is a Grizzly Handler for Grafana alertRuleGroups
+type AlertRuleGroupHandler struct {
+	Provider grizzly.Provider
+}
+
+// NewAlertRuleGroupHandler returns a new Grizzly Handler for Grafana alertRuleGroups
+func NewAlertRuleGroupHandler(provider grizzly.Provider) *AlertRuleGroupHandler {
+	return &AlertRuleGroupHandler{
+		Provider: provider,
+	}
+}
+
+// Kind returns the kind for this handler
+func (h *AlertRuleGroupHandler) Kind() string {
+	return "AlertRuleGroup"
+}
+
+// Validate checks that the uid format is valid
+func (h *AlertRuleGroupHandler) Validate(resource grizzly.Resource) error {
+	data, err := json.Marshal(resource.Spec())
+	if err != nil {
+		return err
+	}
+	var group models.AlertRuleGroup
+	err = json.Unmarshal(data, &group)
+	if err != nil {
+		return err
+	}
+	uid := h.getUID(group)
+	if uid != resource.Name() {
+		return fmt.Errorf("title/folder combination '%s' and name '%s', don't match", uid, resource.Name())
+	}
+	return nil
+}
+
+// APIVersion returns group and version of the provider of this resource
+func (h *AlertRuleGroupHandler) APIVersion() string {
+	return h.Provider.APIVersion()
+}
+
+// GetExtension returns the file name extension for a alertRuleGroup
+func (h *AlertRuleGroupHandler) GetExtension() string {
+	return "json"
+}
+
+const (
+	alertRuleGroupGlob    = "alert-rules/alertRuleGroup-*"
+	alertRuleGroupPattern = "alert-rules/alertRuleGroup-%s.%s"
+)
+
+// FindResourceFiles identifies files within a directory that this handler can process
+func (h *AlertRuleGroupHandler) FindResourceFiles(dir string) ([]string, error) {
```

Comment:
Is this a Grizzly concept, or something new? Same for next func.

Reply:
That's from grizzly. When performing the pull/push flow, grizzly stores resources based on the `ResourceFilePath` function and discovers them using `FindResourceFiles`